src/main.rs

```diff
@@ -1,155 +1,128 @@
-use std::convert::Infallible;
 use std::env;
 use std::net::SocketAddr;
 
-use bytes::Bytes;
+use axum::{
+    extract::{Path, Query},
+    response::Redirect,
+    routing::get,
+    Router,
+};
 use dotenv::dotenv;
-use http_body_util::Full;
-use hyper::header::{HeaderValue, CONTENT_TYPE};
-use hyper::{Request, Response};
-use hyper::service::service_fn;
-use hyper::server::conn::http1;
-use hyper_util::rt::TokioIo;
+use serde::{de::IntoDeserializer, Deserialize};
 use solana_client::rpc_client::RpcClient;
-use solana_sdk::program_pack::Pack;
-use solana_sdk::pubkey::Pubkey;
+use solana_sdk::{program_pack::Pack, pubkey::Pubkey};
 use spl_token::state::Mint;
 use tokio::net::TcpListener;
 
 lazy_static::lazy_static! {
-    static ref HNT_MINT: Pubkey = Pubkey::from_str_const("hntyVP6YFm1Hg25TN9WGLqM12b8TQmcknKrdu1oxWux");
-    static ref MOBILE_MINT: Pubkey = Pubkey::from_str_const("mb1eu7TzEc71KxDpsmsKoucSSuuoGLv1drys1oP2jh6");
-    static ref IOT_MINT: Pubkey = Pubkey::from_str_const("iotEVVZLEywoTn1QdwNPddxPWszn3zFhEot3MfL9fns");
-
     static ref SOLANA_RPC: String = env::var("SOLANA_RPC").unwrap_or_else(|_| "https://api.mainnet-beta.solana.com".to_string());
 }
 
-const HNT_MAX_SUPPLY: f64 = 223_000_000f64;
-const MOBILE_MAX_SUPPLY: f64 = 230_000_000_000f64;
-const IOT_MAX_SUPPLY: f64 = 200_000_000_000f64;
-
 #[tokio::main]
-pub async fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
+async fn main() {
     pretty_env_logger::init();
     dotenv().ok();
 
-    let port: u16 = env::var("PORT").unwrap_or_else(|_| "3000".to_string()).parse().expect("PORT must be a number");
+    let port: u16 = env::var("PORT")
+        .unwrap_or_else(|_| "3000".to_string())
+        .parse()
+        .expect("PORT must be a number");
+
+    let app = Router::new()
+        .route("/api/stats/supply/{token}", get(handle_supply))
+        .fallback(async || Redirect::permanent("https://world.helium.com"));
 
     let addr = SocketAddr::from(([0, 0, 0, 0], port));
-    let listener = TcpListener::bind(addr).await?;
+    let listener = TcpListener::bind(addr).await.unwrap();
```

Comment:
unwraps are bad. At least use expect if you're not willing to return an error

Reply:
addressed in 0c384bc